### DISCUSSION TITLE
Support for  non-file Items in File Explorer

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-theia/theia#13266] [Q&A]
I am in the process of developing a custom view within the File Explorer interface to facilitate the editing of non-file items. This view will include specific functionalities like creating, deleting, and updating such items.

To better illustrate, please refer to the attached screenshot of the outline view. My goal is to implement a similar view within the File Explorer, allowing users to add, delete, and update new items such as 'b,' 'a,' 'v,' and create new nodes at the 'execute' level.

I am seeking guidance on the most effective approach to achieve this. Your insights and recommendations would be greatly appreciated.

![image](https://github.com/eclipse-theia/theia/assets/116638890/1e85b94b-a9fe-4058-80b0-5e4f74f17b45)

**Comment by planger** [comment]
You can implement such a tree view either as a VS Code extension or as a Theia extension. Both have their pros and cons (see https://eclipsesource.com/blogs/2021/03/24/vs-code-extensions-vs-theia-extensions for an in-depth discussion). For your requirements, it seems the VS Code extension API should be sufficiently powerful, so I'd go with this approach in your case.

Please see the [VS Code Treeview API documentation](https://code.visualstudio.com/api/extension-guides/tree-view) for a description on how to create a VS Code extension that provides a tree view similar to what you've shown above. For more info on how to install such an extension then in Theia, please refer to the [Theia documentation on integrating and developing VS Code extensions](https://theia-ide.org/docs/authoring_vscode_extensions/).

I hope this helps! Let us know if you need more info!

**Comment by msujew** [comment]
In addition to what @planger said, if you want to have this as part of the file explorer, [this answer](https://community.theia-ide.org/t/how-to-show-content-of-files-as-children-in-file-explorer/3369/2) might help you.
